Lodgify.Payments.Stripe.Server.IntegrationTests/Shared/BaseIntegrationTest.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System.Net.Http.Headers;
+using System.Text;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using CacheCow.Client;
+using Lodgify.Payments.Stripe.Server.IntegrationTests.Factories;
+using Lodgify.Payments.Stripe.Server.IntegrationTests.Helpers;
+using Microsoft.EntityFrameworkCore;
+using WireMock.Admin.Mappings;
+using WireMock.Client;
+using Xunit;
+
+namespace Lodgify.Payments.Stripe.Server.IntegrationTests.Shared;
+
+[CollectionDefinition(nameof(PaymentsCollection))]
+public class PaymentsCollection : ICollectionFixture<CustomWebApplicationFactory>;
+
+[Collection(nameof(PaymentsCollection))]
+public abstract class BaseIntegrationTest
```

Comment:
Would it be a base class for all tests? Or only these that require Stripe API exposed behind WireMock?
Because of that how about:
 ```suggestion
public abstract class WireMockStripeIntegrationTest
```

Reply:
My idea was make it common for all because it contains helper method like Insert to db, like deserialize response which is not realated with Stripe or WireMock. Then current solution still make sense for you?

Reply:
Hmm you coupled `CustomWebApplicationFactory` with `BaseIntegrationTest` and `CustomWebApplicationFactory` runs the application with wiremock setup as a stripe API. So how do you plan to introduce tests that call real Stripe API?